### PR TITLE
Adds a taste system!

### PR DESCRIPTION
**Pull request by FlattestGuitar**
:cl: FlattestGuitar
rscadd: Adds a taste system! Drink or eat stuff to find out what it tastes like!
/:cl:

How it works: 
- each reagent has an assigned "taste" to it. For the most part they're unique (and as witty as possible), but the simpler reagents will have repeating tastes.
- If a reagent has enough of a percentage share in the whole mixture, it will show up to you as one of the tastes.
- Depending on your species, that threshold might be lowered. Right now Humans and Skrell need at least 20% of a reagent in something to taste it, Vulps and Tajaran need 10% and everyone else needs 15%.  I'm really on the fence about this, but it might be interesting. IPCs can't taste.
- If there is no reagent that passes the threshold (so many reagents none of them makes for at least 15% of the mix) then you'll get a generic "can't pinpoint what you can taste" message.
- You can't taste the same reagent in less than 12 seconds to avoid spam.

If someone wants to go through hell that would be looking at all the taste descriptions my sleep deprived mind has come up with and offer better versions, it would be deeply appreciated.

You can't taste reagents in pills and patches.

### CODE REVIEW

**Comment by scrubmcnoob** [comment]
seems fun

File: code/modules/mob/living/carbon/human/species/station.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	clothing_flags = HAS_UNDERWEAR | HAS_UNDERSHIRT | HAS_SOCKS
 	bodyflags = HAS_SKIN_TONE | HAS_BODY_MARKINGS
 	dietflags = DIET_OMNI
+	taste_sensitivity = TASTE_SENSITIVITY_DULL
```
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/species/station.dm]: See [this line](https://github.com/ParadiseSS13/Paradise/pull/7333/files#diff-aecbf391c938576bbf2e44885a709d7aR48). Why are Humans redefined with dull taste here when it explicitly says that the variable definition (its default value) is the factor of how much reagent needed for a human to taste it?

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 2]
```diff
@@ -428,21 +458,25 @@
...
 	drink_name = "Gibbfloat"
 	drink_desc = "Dr. Gibb with ice cream on top."
+	taste_message = "taste revolution"
 
 /datum/reagent/consumable/drink/pumpkinjuice
 	name = "Pumpkin Juice"
 	id = "pumpkinjuice"
 	description = "Juiced from real pumpkin."
 	color = "#FFA500"
+	taste_message = "puke"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: Does pumpkin really taste that bad?

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 3]
```diff
@@ -128,13 +133,15 @@
 	description = "A tetravalent metalloid, silicon is less reactive than its chemical analog carbon."
 	reagent_state = SOLID
 	color = "#A8A8A8" // rgb: 168, 168, 168
+	taste_message = null
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/misc.dm]: So surge tastes like silicon but silicon tastes like...

**Comment by FlattestGuitar** [comment]
@taukausanake and everyone else who wants to look at the descriptions,

Don't make me think. Just suggest your own thing, pls.

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 4]
```diff
@@ -240,13 +258,15 @@
...
 	color = "#60A584" // rgb: 96, 165, 132
+	taste_message = null
 
 /datum/reagent/colorful_reagent
 	name = "Colorful Reagent"
 	id = "colorful_reagent"
 	description = "It's pure liquid colors. That's a thing now."
 	reagent_state = LIQUID
 	color = "#FFFFFF"
+	taste_message = "colors!"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/misc.dm]: Tastes like all the colors!

File: code/modules/reagents/chemistry/reagents/drinks.dm (hunk 2 above)
Comment (tigercat2000) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: pumpkin is delicious you heathen

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 4 above)
Comment (KasparoVy) [inline code/modules/reagents/chemistry/reagents/misc.dm]: Can't say "the rainbow" without risking copyright, can you?

**Comment by taukausanake** [comment]
@FlattestGuitar I'll go through them again later. If you really don't mind ideas I'll give out some

File: code/modules/reagents/chemistry/reagents/alcohol.dm [hunk 5]
```diff
@@ -95,6 +99,7 @@
 	drink_icon = "absinthebottle"
 	drink_name = "Glass of Absinthe"
 	drink_desc = "The green fairy is going to get you now!"
+	taste_message = "fucking pain"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: heathen, absinthe is delicious

File: code/modules/reagents/chemistry/reagents/alcohol.dm [hunk 6]
```diff
@@ -128,6 +133,7 @@
 	drink_icon = "mojito"
 	drink_name = "Glass of Mojito"
 	drink_desc = "Fresh from Spesscuba."
+	taste_message = "mojito"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: As its a mixed drink, could maybe have its own taste message.

Going to brain dump potential variants for you, ignore/use as you desire:
"an alcoholic breath mint"
"happiness"
"holiday"

File: code/modules/reagents/chemistry/reagents/alcohol.dm [hunk 7]
```diff
@@ -586,6 +624,7 @@
 	drink_icon = "antifreeze"
 	drink_name = "Anti-freeze"
 	drink_desc = "The ultimate refreshment."
+	taste_message = "poor life choices"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: 😆

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 8]
```diff
@@ -146,6 +158,7 @@
 	drink_icon = "banana"
 	drink_name = "Glass of banana juice"
 	drink_desc = "The raw essence of a banana. HONK"
+	taste_message = "banana juice"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: "a bad pun"
"funny"

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 9]
```diff
@@ -176,6 +190,7 @@
 	drink_icon = "glass_brown"
 	drink_name = "Glass of  potato juice"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: feel like fixing the extra space here? :o

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 10]
```diff
@@ -200,6 +216,7 @@
 	color = "#DFDFC7" // rgb: 223, 223, 199
 	drink_name = "Glass of soy milk"
 	drink_desc = "White and nutritious soy goodness!"
+	taste_message = "fake milk"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: "milk and lies"

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 11]
```diff
@@ -43,19 +44,22 @@
...
 	diet_flags = DIET_HERB | DIET_OMNI
+	taste_message = "plant matter"
 
 /datum/reagent/consumable/vitamin
 	name = "Vitamin"
 	id = "vitamin"
 	description = "All the best vitamins, minerals, and carbohydrates the body needs in pure form."
 	reagent_state = SOLID
 	color = "#664330" // rgb: 102, 67, 48
+	taste_message = null
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/food.dm]: "Flintstones... somehow"
"childhood"
"overpriced garbage"

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 12]
```diff
@@ -411,6 +438,7 @@
 	description = "A commercial chemical designed to help older men in the bedroom." //not really it just makes you a giant
 	color = "#ff0000"//strong red. rgb 255, 0, 0
 	var/current_size = 1
+	taste_message = "power!"
```
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/misc.dm]: will the punctuation cause an issue with the existing punctuation in the rest of the message?

**Comment by TravellingMerchant** [comment]
This is a fantastic idea!

File: code/modules/reagents/chemistry/reagents/alcohol.dm (hunk 6 above)
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: I vote for happiness

File: code/modules/reagents/chemistry/reagents/drinks.dm (hunk 8 above)
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: I'd say "honk" but you're already using it somewhere. Doesn't mean you can again though

File: code/modules/reagents/chemistry/reagents/food.dm (hunk 11 above)
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: Flintstones or childhood. Those were never garbage

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 13]
```diff
@@ -562,6 +598,7 @@
 	reagent_state = SOLID
 	color = "#50FF00"
 	addiction_chance = 5
+	taste_message = "cheese...?"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: "cheeeeese...?" Got to throw in that really unsureness

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 14]
```diff
@@ -578,6 +615,7 @@
 	description = "A dish made of mashed beans cooked with lard."
 	reagent_state = LIQUID
 	color = "#684435"
+	taste_message = "burritos"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: "Tastes like beans. Feels like it too."

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 15]
```diff
@@ -590,13 +628,15 @@
 	description = "Bread! Yep, bread."
 	reagent_state = SOLID
 	color = "#9C5013"
+	taste_message = "bread"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: "Tastes like the best thing since sliced- wait."

Don't actually, I just wanted to try making a joke

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 16]
```diff
@@ -660,20 +702,23 @@
...
 	color = "#B4641B"
+	taste_message = "gravy"
 
 /datum/reagent/consumable/beff
 	name = "Beff"
 	id = "beff"
 	description = "An advanced blend of mechanically-recovered meat and textured synthesized protein product notable for its unusual crystalline grain when sliced."
 	reagent_state = SOLID
 	color = "#AC7E67"
+	taste_message = "meat; hopefully"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: Semicolon? Someone back this up please, doesn't look correct to me

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 17]
```diff
@@ -747,16 +795,13 @@
...
-	if(method == INGEST)
-		to_chat(M, "<span class='notice'>That tasted amazing!</span>")
-
 /datum/reagent/cholesterol
 	name = "cholesterol"
 	id = "cholesterol"
 	description = "Pure cholesterol. Probably not very good for you."
 	reagent_state = LIQUID
 	color = "#FFFAC8"
+	taste_message = "heart disease"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: If too much cholesterol in SS13 can actually give people a heart attack I'd say to change this to "heart-attack"

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 18]
```diff
@@ -782,6 +827,7 @@
 	description = "Scrapings of some unknown fungus found growing on the station walls."
 	reagent_state = LIQUID
 	color = "#C87D28"
+	taste_message = "puke"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: Moldy maybe? Pretty sure that's what we had before when you'd cook mushroom pizza before they removed fungus from one of the mushrooms. In fact wasn't that some kind of taste feature for mold alone?

File: code/modules/reagents/chemistry/reagents/food.dm [hunk 19]
```diff
@@ -873,12 +923,13 @@
 	description = "A stimulating ichor which causes luminescent fungi to grow on the skin. "
 	color = "#b5a213"
 	var/light_activated = 0
+	taste_message = "puke"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: Glowy puke maybe?

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 20]
```diff
@@ -341,6 +364,7 @@
 	description = "Oh god it never ends, IT NEVER STOPS!"
 	reagent_state = GAS
 	color = "#D06E27"
+	taste_message = "mexican cuisine"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/misc.dm]: I am slightly offended. Then again I technically eat only Tex-Mex food sooooo

File: code/modules/reagents/chemistry/reagents/misc.dm [hunk 21]
```diff
@@ -520,4 +551,4 @@
 	id = "triplepiss"
 	description = "Ewwwwwwwww."
 	reagent_state = LIQUID
-	color = "#857400"
\ No newline at end of file
+	color = "#857400"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/misc.dm]: So none of these above get any tastes? I mean maybe they aren't used but that sounds like a different problem that needs to be solved

File: code/modules/reagents/chemistry/reagents/pyrotechnic.dm [hunk 22]
```diff
@@ -43,6 +45,7 @@
 	reagent_state = SOLID
 	color = "#673910" // rgb: 103, 57, 16
 	process_flags = ORGANIC | SYNTHETIC
+	taste_message = "metal"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/pyrotechnic.dm]: "Rusty metal". Iron oxide is one of the elements used and as a reagent should be there if stabilized

File: code/modules/reagents/chemistry/reagents/pyrotechnic.dm [hunk 23]
```diff
@@ -56,13 +59,15 @@
...
 	color = "#808080" // rgb: 128, 128, 128
+	taste_message = "sweetness"
 
 /datum/reagent/stabilizing_agent
 	name = "Stabilizing Agent"
 	id = "stabilizing_agent"
 	description = "A chemical that stabilises normally volatile compounds, preventing them from reacting immediately."
 	reagent_state = LIQUID
 	color = "#FFFF00"
+	taste_message = null
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/pyrotechnic.dm]: "stable" or "stabilizing". Unless it really doesn't matter and might not

File: code/modules/reagents/chemistry/reagents/pyrotechnic.dm [hunk 24]
```diff
@@ -125,6 +132,7 @@
 	color = "#000000"
 	metabolization_rate = 0.05
 	penetrates_skin = 1
+	taste_message = "sand"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/pyrotechnic.dm]: "gun powder"? Has to be better than sand

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 25]
```diff
@@ -341,6 +349,7 @@
 	metabolization_rate = 0.1
 	penetrates_skin = 1
 	can_synth = 0
+	taste_message = null
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: "rads" or "radiation"

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 26]
```diff
@@ -510,6 +520,7 @@
 	color = "#CF3600"
 	metabolization_rate = 0.1
 	penetrates_skin = 1
+	taste_message = "almonds"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: They say cyanide smells like almonds but does it _taste_ like them? I mean has anyone reported such before they most likely dropped dead?

Also can Botany grow almonds? I don't recall but they probably should

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 27]
```diff
@@ -566,6 +577,7 @@
 	reagent_state = LIQUID
 	color = "#4141D2"
 	process_flags = ORGANIC | SYNTHETIC
+	taste_message = "<span class='userdanger'>ACID</span>"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: "Tastes like your mouth is dissolving. That can't be good"

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 28]
```diff
@@ -629,6 +641,7 @@
 	reagent_state = LIQUID
 	color = "#7F10C0"
 	can_synth = 0
+	taste_message = null
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Is this a stealth poison? I mean maybe there won't be enough in a food to taste if it had one but asking still. Maybe it doesn't have a taste either way

File: code/modules/reagents/chemistry/reagents/water.dm [hunk 29]
```diff
@@ -138,6 +141,7 @@
 	drink_icon = "glass_red"
 	drink_name = "Glass of Tomato juice"
 	drink_desc = "Are you sure this is tomato juice?"
+	taste_message = "blood"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/water.dm]: Blood but in red italics

File: code/modules/reagents/chemistry/reagents/water.dm [hunk 30]
```diff
@@ -238,6 +242,7 @@
 	description = "Smelly water from a fish tank. Gross!"
 	reagent_state = LIQUID
 	color = "#757547"
+	taste_message = "puke"
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/water.dm]: Fish poop

File: code/modules/reagents/chemistry/reagents/water.dm [hunk 31]
```diff
@@ -373,6 +381,7 @@
 	process_flags = ORGANIC | SYNTHETIC		//Admin-bus has no brakes! KILL THEM ALL.
 	metabolization_rate = 1
 	can_synth = 0
+	taste_message = null
```
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/water.dm]: "admin abuse" :^)
Unless you don't want anyone knowing they are being adminbussed

**Comment by taukausanake** [comment]
Okay I'm done

File: code/modules/reagents/chemistry/reagents/drinks.dm (hunk 2 above)
Comment (Anticept) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: @FlattestGuitar I will thumbs down every one of your PRs if you keep this >:c

File: code/modules/reagents/chemistry/reagents/alcohol.dm (hunk 5 above)
Comment (Anticept) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: It tastes like battery acid.

File: code/modules/reagents/chemistry/reagents/food.dm (hunk 11 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/food.dm]: I'd rather this was something really clear and descriptive OR wasn't there at all, considering how useful they are. I really don't like when you can meta stuff based on some esoteric knowledge of how suddenly tasting Flintstones gives you a more efficient metabolic system. 😉

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 12 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/misc.dm]: Shouldn't break it too badly, but it's generally a poor idea to overdo it.

File: code/modules/reagents/chemistry/reagents/food.dm (hunk 16 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/food.dm]: The system lists things like so: `You can taste X, Y and Z.`. Adding full sentences or proper punctuation to the system will make the messages gibberish.

Though I do think this would do better with a hyphen.

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 21 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/misc.dm]: They taste bitter. Like I imagine they would.

File: code/modules/reagents/chemistry/reagents/pyrotechnic.dm (hunk 22 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/pyrotechnic.dm]: I've zero personal experience, but I believe you wouldn't be able to taste the rust.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 25 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Rads don't taste and I'd rather keep all the toxins rather ambiguous, so as not to give people the ability to check the code for what they drank.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 26 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: a) yes
b) no

File: code/modules/reagents/chemistry/reagents/water.dm (hunk 30 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/water.dm]: ew

File: code/modules/reagents/chemistry/reagents/water.dm (hunk 31 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/water.dm]: I...dunno! Hellwater cookies are used surprisingly often, but maybe a part of the adventure is not knowing?

I'll set it to "admin abuse" for now. For the keks.

File: code/modules/reagents/chemistry/reagents/alcohol.dm (hunk 6 above)
Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: Mojito still has a rather specific and non-face-burning flavor. I'd rather keep it this way.

File: code/modules/mob/living/carbon/human/species/station.dm (hunk 1 above)
Comment (FlattestGuitar) [inline code/modules/mob/living/carbon/human/species/station.dm]: I've changed the relevant comment. Humans don't have to be *the norm* every single time.

I think making them the edge case here might be interesting.

**Comment by AffectedArc07** [comment]
This looks like a fucking great idea

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 32]
```diff
@@ -428,21 +458,25 @@
...
+	taste_message = "caustic puke"
 
 /datum/reagent/consumable/drink/grape_soda
 	name = "Grape soda"
 	id = "grapesoda"
 	description = "Beloved of children and teetotalers."
-	color = "#E6CDFF"
\ No newline at end of file
+	color = "#E6CDFF"
+	taste_message = "grape soda"
```
Comment (Ralta) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: Should taste like grape drank

File: code/modules/reagents/chemistry/reagents/alcohol.dm (hunk 5 above)
Comment (Ralta) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: This should say 'tastes like aniseed and pain' or similar. I quite like absinthe.

File: code/modules/reagents/chemistry/reagents/pyrotechnic.dm (hunk 22 above)
Comment (IK3I) [inline code/modules/reagents/chemistry/reagents/pyrotechnic.dm]: I can assure you rust is quite a potent flavor. There's a reason you can taste it in water without visible discoloration. also of note, we need iron to live, that's why we're so good at tasting it in things. Aluminum, pretty flavorless, tastes like whatever is on it really. That's all that's really in thermite, so rusty metal would be relatively accurate all things considered. And no, I don't go around licking metal in my spare time, but when you work with it for a long time, it eventually finds its way into your mouth.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 26 above)
Comment (IK3I) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Fun fact, you ingest cyanide on a near daily basis. It's only really dangerous in extreme dosages. Note that we're talking extreme in the sense of poisons here though. It'll still only take about a half a teaspoon to cause any real harm, though even then, you'd still probably recover if you got medical treatment.

**Comment by Necaladun** [comment]
Fantastic idea.
If you don't know what various things taste like I'll go buy all the booze and foods in game with server funds to find out.

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 26 above)
Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Random nut allergies for the crew, when?

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 20 above)
Comment (Fox-McCloud) [inline code/modules/reagents/chemistry/reagents/misc.dm]: "goonstation"

**Comment by ModDownloading** [comment]
I support this wholeheartedly, though I would greatly appreciate if you added a "poison" of sorts that has no effects other than masking the flavor of the contents of the rest of the food/drink (which means you can stick some in with a drink to mask the fact that there is ethanol/mercury/initropidril/capulettium-plus in it)

File: code/modules/mob/living/carbon/human/species/station.dm (hunk 1 above)
Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/species/station.dm]: Works for me, thanks for addressing this

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 26 above)
Comment (KasparoVy) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: :rofl: Alaphylactic shock isn't fun

File: code/modules/reagents/chemistry/reagents/drinks.dm (hunk 32 above)
Comment (KasparoVy) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: purp drank

**Comment by KasparoVy** [comment]
@Necaladun DO IT FOR SCIENCE

File: code/modules/reagents/chemistry/reagents/misc.dm (hunk 20 above)
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/misc.dm]: The only time I will ever agree to anything that is Goon

**Comment by taukausanake** [comment]
Just got an amazing idea but not sure how hard it would be to implement. When you gib Ian because I was controlling him and making him eat all the food I _think_ his meat comes up as corgi meat? Would it be possible to add to that instant the taste of "corgi" and have it stand above at whatever level in the food it is in? I'd really like people to know they are eating what was Ian at some point.

Also I didn't notice but was Diona roast in there?

**Comment by FlattestGuitar** [comment]
Unfortunately our food is mostly handled through simple reagents such as "protein" or "plant matter", so it would be impossible to make these tastes more detailed than something really generic. I might tackle that in the future, but I'd like to leave that for another PR.

I'm not sure if a neutral-tasting poison that overpowers other tastes is ever going to get used by anyone. Maybe some sort of a reagent you can make? What would that reagent be? At this point you can just add water to stuff to dilute taste.

There's also functionality for stronger tastes that are able to overpower other ones with really small quantities, but right now I haven't really applied it to anything, as I'm unsure of what would really fit that.

**Comment by pinatacolada** [comment]
Add a toothbrush reagent that makes you hyper sensitive to taste. Death by toothpaste followed by orange juice. And condiment reagents like mayonnaise and guacamole that you can add to food, like the ketchup currently does

**Comment by Regen1** [comment]
>Vulps and Tajaran need 10% and everyone else needs 15%.

Why do vulps and tajaran always get the "best" of everything?

Why can't they for once be the worst? Let them have a duller sense of teste instead of a higher one.

**Comment by FlattestGuitar** [comment]
@Regen1 I was under the impression that dogs and cats have a superior sense of taste to humans, but upon checking on the internets, that appears to be mostly false. 

I'm open to suggestions as to how to alter the PR to make for the most interesting combination there.

**Comment by SamHPurp** [comment]
@FlattestGuitar Is there anything to reduce duplicates? Lets say I make a (digusting) sillicon and aluminum cocktail, it'll tell me it tastes like "metal and metal".

**Comment by FlattestGuitar** [comment]
@SamHPurp yep. If two reagents share the same taste string, they will merge into a single one during the calculation, essentially. Means they can reach the thresholds together and will only show up once during tasting.

**Comment by SamHPurp** [comment]
@FlattestGuitar ah nice, great work.

File: code/modules/reagents/chemistry/reagents/food.dm (hunk 11 above)
Comment (taukausanake) [inline code/modules/reagents/chemistry/reagents/food.dm]: If you ever had Flintstones you'd understand

File: code/modules/reagents/chemistry/reagents/water.dm [hunk 33]
```diff
@@ -346,6 +353,7 @@
 	description = "Something that shouldn't exist on this plane of existance."
 	process_flags = ORGANIC | SYNTHETIC //ethereal means everything processes it.
 	metabolization_rate = 1
+	taste_message = null
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/water.dm]: damnation?

File: code/modules/reagents/chemistry/reagents/toxins.dm (hunk 26 above)
Comment (Rumiluntti) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Cyanide smells and tastes faintly of bitter-ish almonds.
Don't ask me why I know this.

**Comment by Anticept** [comment]
@Regen1 i forgot to reply to this.

Funny enough, cats and dogs actually have far worse vision than humans resolution wise. 5-6 times worse. We can see much farther than they can. However, they have superior night vision.

**Comment by outbools** [comment]
@FlattestGuitar people are able to taste reagents via patches and thats look inconsistent

**Comment by FlattestGuitar** [comment]
You can no longer taste pills or patches.

File: code/modules/mob/living/carbon/carbon.dm [hunk 34]
```diff
@@ -1006,7 +1006,7 @@ var/list/ventcrawl_machinery = list(/obj/machinery/atmospherics/unary/vent_pump,
 
 /*TO DO - If/when stomach organs are introduced, override this at the human level sending the item to the stomach
 so that different stomachs can handle things in different ways VB*/
-/mob/living/carbon/proc/consume(var/obj/item/weapon/reagent_containers/food/toEat, var/bitesize_override)
+/mob/living/carbon/proc/consume(var/obj/item/weapon/reagent_containers/food/toEat, var/bitesize_override, var/taste = 1)
```
Comment (lordpidey) [inline code/modules/mob/living/carbon/carbon.dm]: `var/taste = TRUE` should be used instead.

File: code/modules/reagents/chemistry/reagents/toxins.dm [hunk 35]
```diff
@@ -247,10 +255,6 @@
 	if(method == INGEST)
 		if(ishuman(M))
 			var/mob/living/carbon/human/H = M
-
-			if(volume < 10)
-				to_chat(M, "<span class='danger'>The greenish acidic substance stings you, but isn't concentrated enough to harm you!</span>")
```
Comment (tigercat2000) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: Why did you remove this?

Comment (FlattestGuitar) [inline code/modules/reagents/chemistry/reagents/toxins.dm]: You can already taste the acid with a massive taste message, so two different outputs on the taste seem redundant to me.

**Comment by FlattestGuitar** [comment]
Oh ffs git

File: code/modules/mob/living/carbon/carbon.dm (hunk 34 above)
Comment (tigercat2000) [inline code/modules/mob/living/carbon/carbon.dm]: Please resolve this.

File: code/modules/mob/living/carbon/human/species/station.dm (hunk 1 above)
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/species/station.dm]: Humans are generally considered the base average of species- They should really just use the normal taste sensitivity.

File: code/modules/reagents/chemistry/reagents/drink_base.dm [hunk 36]
```diff
@@ -23,4 +23,4 @@
 	if(adj_temp_cool)
 		if(M.bodytemperature > 310)//310 is the normal bodytemp. 310.055
 			M.bodytemperature = max(310, M.bodytemperature - (adj_temp_cool * TEMPERATURE_DAMAGE_COEFFICIENT))
-	..()
\ No newline at end of file
+	..()
```
Comment (tigercat2000) [inline code/modules/reagents/chemistry/reagents/drink_base.dm]: The fact that you added a newline here makes me assume you haven't disabled the stupid automatic behaviour in Atom over it; You should really go to `Settings > Packages > Whitespace` and uncheck `Ensure Single Trailing Newline`.

File: code/modules/reagents/chemistry/reagents/drinks.dm [hunk 37]
```diff
@@ -160,6 +173,7 @@
 	drink_icon = "nothing"
 	drink_name = "Nothing"
 	drink_desc = "Absolutely nothing."
+	taste_message = "nothing...how?"
```
Comment (tigercat2000) [inline code/modules/reagents/chemistry/reagents/drinks.dm]: Space after ellipses please.

File: code/modules/reagents/chemistry/reagents/drugs.dm [hunk 38]
```diff
@@ -439,6 +439,7 @@
 	reagent_state = LIQUID
 	color = "#644600"
 	addiction_chance = 30
+	taste_message = "puke...or worse"
```
Comment (tigercat2000) [inline code/modules/reagents/chemistry/reagents/drugs.dm]: Spaces after ellipses please.

File: code/modules/reagents/chemistry/reagents/alcohol.dm [hunk 39]
```diff
@@ -9,6 +9,7 @@
 	can_grow_in_plants = 0	//Alcoholic drinks won't be grown in plants (would "water down" random seed chems too much)
 	var/dizzy_adj = 3
 	var/alcohol_perc = 1 //percentage of ethanol in a beverage 0.0 - 1.0
+	taste_message = "liquid fire"
```
Comment (Crazylemon64) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: is that a nethack reference :eyes:

Comment (SamHPurp) [inline code/modules/reagents/chemistry/reagents/alcohol.dm]: we need more nethack references

**Comment by KasparoVy** [comment]
Ayy, finally! Nice.
